modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/internal/data/handler/test/BatchEnginePortletDataHandlerTest.java

```diff
@@ -474,6 +485,270 @@ public void testExportImportPrivateLayoutsToOtherSite() throws Exception {
 		Assert.assertNull(layout2);
 	}
 
+	@Test
+	public void testExportImportRootModelHierarchy() throws Exception {
+		Group group = _stagingGroupHelper.fetchCompanyGroup(
+			TestPropsValues.getCompanyId());
+
+		DefaultObjectEntryManager defaultObjectEntryManager =
+			(DefaultObjectEntryManager)_objectEntryManager;
+
+		DTOConverterContext dtoConverterContext =
+			new DefaultDTOConverterContext(
+				false, Collections.emptyMap(), dtoConverterRegistry, null,
+				LocaleUtil.getDefault(), null, TestPropsValues.getUser());
+
+		ObjectDefinition objectDefinitionA = _addAndPublishObjectDefinition(
+			"A");
+
+		ObjectDefinition objectDefinitionAA = _addAndPublishObjectDefinition(
+			"AA");
+
+		ObjectDefinition objectDefinitionAAA = _addAndPublishObjectDefinition(
+			"AAA");
+
+		ObjectDefinition objectDefinitionAAAA = _addAndPublishObjectDefinition(
+			"AAAA");
+
+		ObjectDefinition objectDefinitionAAAAA = _addAndPublishObjectDefinition(
+			"AAAAA");
+
+		ObjectDefinition objectDefinitionB = _addAndPublishObjectDefinition(
+			"B");
```

Comment:
Can you take a look at how the root-model tests are written in `ObjectDefinitionTreeUtilTest` and check what `TreeTestUtil` and `ObjectRelationshipTestUtil` provide that you can reuse in your test?

For example:
- review how the tree is created (`TreeTestUtil.createObjectDefinitionTree()`) and how entries are added so you can apply the same patterns here.
- variable name pattern